### PR TITLE
Golint and Ineffassign  fixes

### PR DESCRIPTION
**Pull request by thaJeztah**
Saw these on https://goreportcard.com/report/github.com/docker/docker#golint

ping @dnephin @vdemeester

### CODE REVIEW

**Review by boaz0 — APPROVED** [review]
LGTM 👍 
GJ

File: api/server/middleware/version.go [hunk 1]
```diff
@@ -57,8 +57,8 @@ func (v VersionMiddleware) WrapHandler(handler func(ctx context.Context, w http.
 		if versions.GreaterThan(apiVersion, v.defaultVersion) {
 			return versionUnsupportedError{version: apiVersion, maxVersion: v.defaultVersion}
 		}
-		// nolint: golint
-		ctx = context.WithValue(ctx, "api-version", apiVersion)
+		type contextKey string
+		ctx = context.WithValue(ctx, contextKey("api-version"), apiVersion)
```
Comment (thaJeztah) [inline api/server/middleware/version.go]: Looks like it didn't like this change; fixing

**Review by vdemeester — APPROVED** [review]
LGTM 🐯

File: api/server/httputils/httputils.go [hunk 2]
```diff
@@ -6,14 +6,12 @@ import (
...
 	"strings"
 
+	"github.com/docker/docker/api/server/middleware"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	"golang.org/x/net/context"
 )
 
-// APIVersionKey is the client's requested API version.
-const APIVersionKey = "api-version"
```
Comment (thaJeztah) [inline api/server/httputils/httputils.go]: moved this to `middleware/version.go` as that's where this key is actually set, so it seemed more logical to define the name there 😅

**Review by yongtang — APPROVED** [review]
LGTM

**Pull request merged**